app/forms/ctc/direct_deposit_form.rb

```diff
@@ -9,5 +9,12 @@ class DirectDepositForm < QuestionsForm
     def save
       @intake.update(bank_account_attributes: attributes_for(:bank_account))
     end
+    
+    def self.existing_attributes(intake)
+      return super unless intake.bank_account.present?
+
+      # bank_name is encrypted, but we want it to be editable for clients
```

Comment:
I don't think this is because the attr is encrypted, but because it is on a sub-model; the form is dealing only with the Intake right now so it won't read the existing value of `bank_name` (or `account_type`) unless we copy it over onto the form explicitly.

There should be some more general way to do this (maybe `fields_for :bank_account` in the form) but I haven't quite got it to work in experimenting so maybe this is fine for now (with a more accurate comment)

Reply:
if we keep the explicit merge of `bank_account` fields you may want to add `account_type` to it so it is still populated when someone goes back to edit

Reply:
ah, so its a combo -- because of the way "existing attributes" works, it only slices off "real" attributes, so the encrypted ones always need to be explicitly set (wondering whether account_name even needs to be encrypted though if we want it to be editable), but also because they aren't on intake, all of the attrs need to be set. I had a prev implementation that overwrote from_intake that dealt with account_type properly, but yes -- i think this will not preserve account type properly.

Reply:
yah I didn't realize in the last comment that `intake.attributes` isn't gonna give any of the encrypted stuff :)

---

For controllers/forms that pull things off a submodel like `bank_account` I'm trying to imagine if there's a better implementation for `FormAttributes.existing_attributes` that handles submodel attributes by default

Like whenever we say a form has `attributes_for :bank_account, ...` the base `existing_attributes` method could see if `:bank_account` was a real association on the main model and try to read attributes from it onto the base form

then we could also replace calls like `model.attributes` with something like

`scoped_attributes[model.table_name].map { |attr| model.send(attr) }`

(I wish there was a more succinct way to write that) to handle anything with fancy getters such as encrypted fields

---

though having dived deeper into `FormAttributes` now I'm a little creeped out by how it attempts to flatten a bunch of things that potentially live on subobjects into one flat list of form attributes which then get unflattened on `save`. but maybe that's just something I need to adapt to